readers/combined.cc

```diff
@@ -114,6 +116,10 @@ class mutation_fragment_merger {
                     }
                     return stop_iteration::no;
                 } else {
+                    if (_statistics && begin->fragment.is_clustering_row()) {
+                        ++_statistics->rows_merged_histogram[fragments.size()];
+                    }
```

Comment:
maybe we should treat the caller resizing the vector upfront as optimization and optionally grow the histogram here (slower but still works). today it works, but if we have another user tomorrow that forgets to resize the vector (or the resizing in the current user goes away by mistake), we won't have surprises. Usually it's better to not have such contracts between caller and callees. if we go that way, I suggest hiding it in a function to avoid polluting this main function.

Reply:
Agree, we should make sure the code is safe, the pre-set size is an optimization, let's not rely on it.

Reply:
The counter is increased in `mutation_fragment_merger` which does not have any knowledge on the number of sstable files involved. Thus I cannot properly resize the vector there even though I would really like.

The `compation` class (the caller) that requests the statistics to be collected has this knowledge and should ensure the vector is of a proper size.

If we want to check for the out-of-bound index and optionally resize the vector, then I insist to take a step back and use a hashmap. It does it for us and we can still do some optimizations by calling `reserve` on it to avoid unnecessary allocations.

Reply:
I think it's okay to require the caller and callee to cooperate here. This is going to be called hundreds of thousands of times per second in some cases, we can't start hashing things in hot paths like that for no reason.